apps/labs/posts/making-gpus-accessible-to-pydata-ecosystem-via-array-api.md

```diff
@@ -0,0 +1,522 @@
+This blog post is only meant to illustrate how we can use it to make GPUs
+accessible to users of libraries like SciPy and scikits and the performance
+benefits of that. It does not necessarily define any best practices around
+adopting the standard. Such adoption best practices are still a work in
+progress and will evolve over the next year.
+
+## The example: image segmentation through spectral clustering
+
+The example we have chosen to demonstrate adopting the Array API standard to make
+GPUs accessible in SciPy and scikits is the one taken from scikit-learn’s
+documentation: segmenting a picture of Greek coins into regions. This example
+uses spectral clustering on a graph created from voxel-to-voxel difference on
+an image to break this image into multiple partly-homogeneous regions (or clusters).
+In essence spectral clustering is about finding the clusters in a graph, i.e.,
+finding the mostly connected subgraph and thereby identifying clusters.
+Alternatively, it can be described as finding subgraphs having a maximum number
+of within-cluster connections and a minimum number of between-cluster
+connections.
+
+<p align="center">
+    <img
+     alt="Greek coins from Pompeii"
+     src="/posts/making-gpus-accessible-to-pydata-ecosystem-via-array-api/greek_coins_original.jpeg" />
+</p>
+<p align="center">
+<i>Greek coins from Pompeii ("British Museum, London, England").<br />
+Source: <a href="https://www.brooklynmuseum.org/opencollection/archives/image/51611">www.brooklynmuseum.org/opencollection/archives/image/51611</a></i>
+</p>
+
+The dataset we are using for this demonstration is of greek coins from Pompeii
+(from the `skimage.data.coins` module). This is a data of image of several
+coins outlined against a gray background.
+
+We convert the data of the coins into a graph object and then apply spectral
+clustering to it for segmenting coins.
+
+## The process for making the code run on GPU
+
+#### `get_namespace` - getting the array API module
+
+[NEP 47](https://numpy.org/neps/nep-0047-array-api-standard.html), "Adopting
+the array API standard", outlines a basic workflow for adopting the array API
+standard for array provider and consumer libraries. Now we will describe
+in detail about how we went through the process of implementing this in forks
+of SciPy, scikit-learn and scikit-image.
+
+The Array API defines a method named `__array_namespace__` which returns
+the array API module with all the array API functions accessible from it.
+It has a couple of key benefits:
+
+- for the array consumer libraries to be able to check if the incoming array
+  supports the standard.
+- to access all the functions that work with the array object without having
+  the need to import the Python module explicitly.
+
+NEP 47 recommends defining a utility function named `get_namespace` to check
+for such an attribute.
+
+An example of potential implementation of `get_namespace` function is defined as follows:
+
+```python
```

Comment:
The code blocks have syntax highlighting, but the highlighting has contrast accessibility issues. Note to self to create GitHub issue about this.